modules/programs/hexchat.nix

```diff
@@ -0,0 +1,262 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
+  meta.maintainers = [ maintainers.superherointj ];
+
+  options.programs.hexchat = {
+    channels = with lib.types;
+      let
+        modChannelOption = submodule {
+          options = {
+            autojoin = mkOption {
+              type = nullOr (listOf str);
+              description =
+                "Channels list to autojoin on connecting to server.";
+              example = ''[ "##linux" "#nix" ]'';
+              default = [ ];
+            };
+            charset = mkOption {
+              type = str;
+              description = "Charset";
+              default = "";
+              example = "UTF-8 (Unicode)";
+            };
+            commands = mkOption {
+              type = nullOr (listOf str);
+              description = "Commands to be executed on connecting to server.";
+              example = ''[ "ECHO Greetings fellow Nixer! ]'';
+              default = [ ];
+            };
+            loginMethod = mkOption {
+              type = with lib.types;
```

Comment:
Remove it (it is already at top).

Reply:
Sorry. I don't understand what you mean. I don't see any duplication of "loginMethod" and defaults are currently necessary.

Reply:
The default field with empty list as value?

Reply:
No, the `with types`. It is already here: https://github.com/nix-community/home-manager/pull/1724/files#diff-a1c5686a4f891b3c24d41e6624397f028b25863272dd0a9acb4975b9663554b9R9

Reply:
Done.

Reply:
Done.